Source/items.h

```diff
@@ -348,6 +349,20 @@ struct ItemStruct {
 			return false;
 		}
 	}
+
+	UiFlags getTextColor() const
+	{
+		if (!_iStatFlag)
+			return UIS_RED;
+		switch (_iMagical) {
+		case ITEM_QUALITY_MAGIC:
+			return UIS_BLUE;
+		case ITEM_QUALITY_UNIQUE:
+			return UIS_GOLD;
+		default:
+			return UIS_SILVER;
+		}

```

Comment:
Random question: is there no pattern matching proposal for C++? This would look so much more clean:
```csharp
return item switch
{
    { _iStatFlag: false } => UIS_RED,
    { _iMagical: ITEM_QUALITY_MAGIC } => UIS_BLUE,
    { _iMagical: ITEM_QUALITY_UNIQUE } => UIS_GOLD,
    _ => UIS_SILVER,    
};
```

Reply:
Yes there is a proposal like that